README.rst

```diff
@@ -990,6 +992,106 @@ so they can be overridden if needed.
+
+7. After clicking on **Save and continue editing**, you will see that OpenWISP
+   has automatically created public and private key for WireGuard server in
+   **System Defined Variables** along with internal IP address information.
+
+.. image:: https://raw.githubusercontent.com/openwisp/openwisp-controller/wireguard-vxlan/docs/wireguard-tutorial/vpn-server-3.png
+   :alt: WireGuard VPN server configuration example 3
+
+2. Deploy Wireguard VPN Server
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you haven't already setup WireGuard on your VPN server, this will be a good
+time do so. We stress on using `ansible-wireguard-openwisp <https://github.com/openwisp/ansible-wireguard-openwisp>`_
+role for installing WireGuard since it also installs scripts that allows
+OpenWISP to manage WireGuard VPN server.
+
+Pay attention to the VPN server attributes used in your playbook. It should be same as
+VPN server configuration in OpenWISP.
+
+3. Create VPN template for WireGuard VPN Server
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. Visit ``/admin/config/template/add/`` to add a new template.
+2. Set ``Wireguard Client`` as **Name** (you can set whatever you want) and
+   select ``VPN-client`` as **type** from the dropdown list.
+3. The **Backend** field refers to the backend of the device this template can
+   be applied to. For this example, we will leave it to ``OpenWRT``.
+4. Select the correct VPN server from the dropdown for the **VPN** field. Here
+   it is ``Wireguard``.
+5. Ensure that **Automatic tunnel provisioning** is checked. This will make
+   OpenWISP to automatically generate public and private keys and provision IP
+   address for each WireGuard VPN client.
+6. After clicking on **Save and continue editing** button, you will see details
+   of *Wireguard* VPN server in **System Defined Variables**. The template
+   configuration will be automatically generated which you can tweak
+   accordingly. We will use the automatically generate VPN client configuration
+   for this example.
+
+.. image:: https://raw.githubusercontent.com/openwisp/openwisp-controller/wireguard-vxlan/docs/wireguard-tutorial/template.png
+    :alt: WireGuard VPN client template example
+
+4. Apply Wireguard VPN template to devices
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**Note**: This step assumes that you already have a device registered on
+OpenWISP. Register or create a device before proceeding.
+
+1. Open the **Configuration** tab of the concerned device.
+2. Select the *WireGuard Client* template.
+3. Upon clicking on **Save and continue editing** button, you will see some
+   entries in **System Defined Variables**. It will contain internal IP address,
+   private and public key for the WireGuard client on the device along with
+   details of WireGuard VPN server.
+
+.. image:: https://raw.githubusercontent.com/openwisp/openwisp-controller/wireguard-vxlan/docs/wireguard-tutorial/device-configuration.png
```

Comment:
reminder: we may have to update the screenshot if we change the public key variable name (pubkey > public_key).